docs/development.md

```diff
@@ -1,32 +1,136 @@
+
+In either case, when choosing a Linux distribution, Ubuntu is fine.
+
+### Connecting to localhost
+
+You will need edit your system's HOSTS file so that when you navigate to https://www.wikijump.test, your browser knows that you want to connect to your own virtual server and not somewhere on the internet. On Linux, your HOSTS file is `/etc/hosts`. On Windows, it is `C:\Windows\System32\drivers\etc\hosts`, but in order to edit it you will need to be running an editor (e.g. Notepad) as administrator. If you are running WSL2, edit your Windows HOSTS file; ignore your Linux HOSTS file.
+
+Add the following lines to your HOSTS file:
+
+```
+127.0.0.1 localhost
+127.0.0.1 wikijump.test
+127.0.0.1 www.wikijump.test
+127.0.0.1 profiles.wikijump.test
+127.0.0.1 template-en.wikijump.test
+127.0.0.1 wjfiles.test
+127.0.0.1 www.wjfiles.test
+```
+
+You will also need to add a line for any wiki that you create, e.g. `127.0.0.1 my-new-wiki.wikijump.test`.
+
+## Installation
+
 The `installer` folder has everything you need to run a local Wikijump install either in a container or on metal or a VM.
 
-### Installation
+### Installation via [Docker](https://www.docker.com/) locally
+
+The recommended way to install Wikijump is via Docker. Docker is a container system that lets you easily create and destroy different Wikijump builds, and it also acts like a sandbox to protect the rest of your system from dependency pollution.
+
+You will need Docker installed and running:
+
+<table>
+  <thead><tr><th>Ubuntu / Ubuntu VM</th><th>Windows via WSL2</th></tr></thead>
+  <tbody valign="top"><tr>
+    <td>
+      <p><pre># apt install docker</pre></p>
+      <p>Start the Docker daemon in another terminal, and leave it running in the background:</p>
+      <p><pre>$ dockerd</pre></p>
+    </td>
+    <td>
+      <p><pre># apt install docker</pre></p>
+      <p>Install <a href="https://docs.docker.com/docker-for-windows/install-windows-home">Docker Desktop</a>, which you'll be using as the Docker daemon, and leave it running in the background.</p>
+      <p>If you have set <code>appendWindowsPath=false</code> in your WSL config, then you may hit an error along the lines of <code>"docker-credential-desktop.exe": executable file not found in $PATH</code>. In this case you should either add <code>/mnt/c/Program\ Files/Docker/Docker/resources/bin</code> to your PATH, or <a href="https://github.com/rossjrw/dotfiles/blob/3c5445abb138b735cc3caf61f070c9125fa87d2f/.profile#L28">create a symlink</a> in a directory that is in your PATH already.</p>
+    </td>
+  </tr></tbody>
+</table>
 
-You can create a docker container using the following:
+The first step is to build the Docker image, which is a single source of information for Docker to use in a container later:
 
 ```bash
-$ docker build . -t scpwiki/wikijump:local
-$ docker create --name wikijump -p 80:80 scpwiki/wikijump:local
-$ docker start wikijump
+$ cd installer
+$ docker build . -t scpwiki/wikijump:local --no-cache
```

Comment:
Do we want to necessarily tell people to use `--no-cache` every time? It really speeds up builds if you're doing it a lot

Reply:
I'm not really sure - I've found that if I've changed something and don't add `--no-cache`, it just rebuilds the same thing without my changes. I'd argue that it's better to add the option and give the reader the option of omitting it, rather than them potentially not knowing what to do if they do need it.